cirrus-geo/cirrus-geo
# Initial work for v1

This PR is all a starting point and we'll need to do more, but it includes:

* clean up all code no longer required with the new operational paradigm
  * `cirrus.cli` code combed over for what is relevant to the cirrus management cli and moved there, the rest removed
  * `cirrus.core` code removed
  * `cirrus.test` module removed
* reorganize remaining code
  * `cirrus.lib2` becomes `cirrus.lib`
  * `cirrus.builtins` becomes `cirrus.lambda_handlers` and flattened
* switch to `pyproject.toml` and `pip-compile` for deps
  * though I am uncertain how we should manage versions given that we _do_ want to be able to install this package into a virtual env with pip from pypi (i.e., requirements files alone ain't going to cut it)
* pull in `cirrus-mgmt` cli code and install as `cirrus` console script
  * code from https://github.com/cirrus-geo/cirrus-mgmt/pull/10 where I made the cli standalone and implemented a PoC for how to manage config from AWS resources
* docs dependency on `cirrus-docs` removed in favor of plain sphinx build and docs sources moved into `docs/`
* I also looked for additional dead code that wasn't necessarily dead just from this work and removed it.
* I pulled out all the backwards-compatibility code for deprecated features that I could find:
  * `output_options` support
  * support for the `FAILED` and `INVALID` SNS topics
  * removed `InvalidInput` exception in _this_ `cirrus.lib`
  * `legacy` API support
* I had concerns about recent changes to `BatchHandler` and the subclasses that were created, `SNSPublisher` and `SQSPublisher`. I took this opportunity to address those concerns.
